conans/test/integration/options_test.py

```diff
@@ -64,3 +64,59 @@ class ConanLib(ConanFile):
                       client.user_io.out)
         conaninfo = load(os.path.join(client.current_folder, CONANINFO))
         self.assertNotIn("zlib:shared=True", conaninfo)
+
+    def general_scope_options_test(self):
+        # https://github.com/conan-io/conan/issues/2538
+        client = TestClient()
+        conanfile_libA = """
+from conans import ConanFile
+
+class LibA(ConanFile):
+    name = "libA"
+    version = "0.1"
+    options = {"shared": [True, False]}
+    default_options= "shared=False"
+
+    def configure(self):
+        self.output.info("shared=%s" % self.options.shared)
+    """
+        conanfile_libB = """
+from conans import ConanFile
+
+class LibB(ConanFile):
+    name = "libB"
+    version = "0.1"
+    options = {"shared": [True, False]}
+    default_options= "shared=False"
+    requires = "libA/0.1@danimtb/testing"
+
+    def configure(self):
```

Comment:
This test looks a bit complex for the use case. Just defining ``-o *:shared`` in the command line should be enough to check it, wouldn't it?

Reply:
No. this is in fact the real issue this is testing. Read the description in the PR comments